CMakeLists.txt

```diff
@@ -659,7 +659,7 @@ if(UNIX)
     elseif(EMSCRIPTEN)
       # no need to link to system libs with emscripten
     elseif(QNXNTO)
-      # no need to link to system libs with QNX
+      set(OPENCV_LINKER_LIBS ${OPENCV_LINKER_LIBS} m)
```

Comment:
@pablorcum @blackliner Could you please check merged QNX-related changes? (PR #20399)

Reply:
LGTM: https://www.qnx.com/developers/docs/6.4.1/neutrino/lib_ref/c/cos.html

Reply:
I've built it successfully for QNX after this change.